livekit-agents/livekit/agents/llm/chat_context.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2023 LiveKit, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import enum
+from typing import Union
+
+from attrs import define
+from livekit import rtc
+
+
+class ChatRole(enum.Enum):
+    SYSTEM = "system"
+    USER = "user"
+    ASSISTANT = "assistant"
+    TOOL = "tool"
+
+
+@define
+class ChatMessage:
+    role: ChatRole
+    text: str
+    images: list[Union["ChatMessageVideoFrameImage", "ChatMessageURLImage"]] = []
+
+
+@define
+class ChatContext:
+    messages: list[ChatMessage] = []
+
+
+@define
+class ChatMessageVideoFrameImage:
```

Comment:
Not fan of the long names here what about smthg like:
```
@define
class ChatImage:
   frame: url.URL | str | rtc.VideoFrame
   detail: Literal["low", "high", "auto"]
   
```

Reply:
yeah I went back and forth on this. I'll go back to it. I won't have detail there though because that might be LLM specific.